### DISCUSSION TITLE
Typescript error/warning when using register inside the field (useFieldArray)

### DISCUSSION THREAD

**Discussion by matusca96** [react-hook-form/react-hook-form#4877] [Q&A]
Hello, guys, that's my first topic in whole github, and I have a kind of problem when using register inside a field with the useFieldArray hook.

Actually this doesn't affect my project directly, everything is working fine, but in my VSCode, the typescript shows an error in the register line for each field in my array:

`{...register(`perguntaItens.${index}.opcao` as const)}`

The code above is one of my array fields, they are equal and all of them return the same error below:

Argument of type '`perguntaItens.${number}.opcao`' is not assignable to parameter of type '`${string}` | `${string}.${string}` | `${string}.${number}`'.
  Type '`perguntaItens.${number}.opcao`' is not assignable to type '`${string}.${number}`'.ts(2345)

As I said, besides this error, my application is building normally, and my values are being returned very well:

![image](https://user-images.githubusercontent.com/24362309/115399665-d4253b00-a1be-11eb-936b-260f78d724af.png)

And another problem, everytime that I create a new entry using append(), my devtools display this warning:

react_devtools_backend.js:2557 Warning: flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering. Consider moving this call to a scheduler task or micro task.

Someone to help? Thansk a lot!

Regards.

**Comment by bluebill1049** [comment]
1. The first issue is related to TS, I think it will be released soon (as it's already merged) to be solved.
2. append(), we don't support empty arg for append

**Reply by matusca96** [reply]
I'm sorry, my append is like this:

`append({
  opcao: "",
  preco: 0,
  ordem: fields.length+1,
  exibir: true
})`

I was thinking if the problem was the "ordem: fields.length+1" but I've tried to start this field with 0, but with no success, the flushSync error remains.

**Reply by bluebill1049** [reply]
can you share a simple and reproduceable csb?
